sdk/identity/azure-identity/migration_guide.md

```diff
@@ -0,0 +1,58 @@
+# Guide for migrating to azure-identity from azure-common
+
+## JSON- and file-based authentication
+
+To encourage best security practices, `azure-identity` does not support JSON- and file-based authentication in the same
+way as `azure-common`. `azure-common` provided factory methods like [`get_client_from_json_dict`][client_from_json] and
+[`get_client_from_auth_file`][client_from_auth_file] that are no longer available in `azure-identity`.
+
+In `azure-common` you could provide credentials in a JSON dictionary, or from a JSON file:
+```python
+from azure.common.client_factory import get_client_from_json_dict, get_client_from_auth_file
+from azure.mgmt.keyvault import KeyVaultManagementClient
+# Provide credentials in JSON:
+json_dict = {
+    "clientId": "...",
+    "clientSecret": "...",
+    "subscriptionId": "...",
+    "tenantId": "...",
+    "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+    "resourceManagerEndpointUrl": "https://management.azure.com"
+}
+client = get_client_from_json_dict(KeyVaultManagementClient, json_dict)
+# Or, provide credentials from a JSON file:
+client = get_client_from_auth_file(KeyVaultManagementClient, "credentials.json")
+```
+
+If it's too difficult to migrate away from file-based authentication despite the security risks, you can still
```

Comment:
I might be biased and not native speaker, but I alsmost feel it as passive-aggressive "If it's too difficult... despite the security risks" => "Really you could try harder" :p

What about:
If there is any reasons that makes the migration impossible to achieve immediately, you can still use.....

Reply:
That's a good point. I meant to just emphasize that using this workaround isn't recommended if it can be avoided, but I can see how it could be read as passive-aggressive